src/main/java/org/opensearch/security/dlic/rest/api/MigrateApiAction.java

```diff
@@ -102,6 +103,56 @@ private void migrateApiRequestHandlers(RequestHandler.RequestHandlersBuilder req
         requestHandlersBuilder.allMethodsNotImplemented().override(Method.POST, (channel, request, client) -> migrate(channel, client));
     }
 
+    protected boolean updateSecurityIndex(
+        final RestChannel channel,
+        final Client client,
+        ImmutableList.Builder<SecurityDynamicConfiguration<?>> builder
+    ) {
+        final List<SecurityDynamicConfiguration<?>> dynamicConfigurations = builder.build();
+        if (dynamicConfigurations.isEmpty()) {
+            LOGGER.info("Nothing to migrate");
+            return false;
```

Comment:
This needs to send a response to the RestChannel. Otherwise, the called will get an ever pending request.

Reply:
I changed it slightly for the case you are pointing out here. The function now returns a boolean which indicates if anything was updated or not. If nothing was updated it returns false and then [down below](https://github.com/opensearch-project/security/pull/5215/files#diff-a8428898b755f3a54f89dd5a5c2ad8868b4d5cb549aab30ce252e00040c0c5aaR181) it will respond on the channel.

I did it this way to work in both cases where all configs or updated or just the configs that are missing the `_meta` header.

Reply:
Ah, okay.